### ISSUE TITLE
Training own Dataset on Windows 10?

### ISSUE THREAD

**Issue by WhyWouldYouTilt** [matterport/Mask_RCNN#1891]
Hey,

so i started implementing Mask RCNN for my own Dataset. I followed the balloon splash tutorial and right now im at the point where i implemented the functions to load my Data and show Masks/Bounding boxes etc. (everything of the first jupyter notebook is working)
I got a Dataset of around 500 pictures with annotations and want to train on the pretrained coco net, but i am not quite sure how to do it. 
My GPU is: NVIDIA GTX 1060 with 3GB RAM
The tutorial says, that if you dont have a GPU with 12GB RAM it is not really possible to train it, so he(in the tutorial) used amazons P2 instances.

I wanted to do that too, but searching on the website i found, that this only works for Linux systems?
I dont really know how to train my net for my data, so it would be really nice if someone of you could help me out...

Best regards
WhyWouldYouTilt

**Comment by marcfielding1** [comment]
It should work, you don't need a GPU with that memory I did the balloon example on a GTX 1070 with 6gb of available memory in 20 minutes(rather than a p2 which takes an hour).

What makes you believe it won't work on windows?

**Comment by marcfielding1** [comment]
If you follow the balloon example, load your annotations from however you've produced them then you should be able to do it fine, if you could tell us what annotations you're using, with an example format and example code of your class I'm sure we can figure it out.

**Comment by WhyWouldYouTilt** [comment]
When i try to train the balloon example i get these OOM errors:(I already set the IMAGES_PER_GPU=1)

ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[200,256,28,28] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
         [[Node: mrcnn_mask_deconv/conv2d_transpose = Conv2DBackpropInput[T=DT_FLOAT, _class=["loc:@train...propFilter"], data_format="NCHW", dilations=[1, 1, 1, 1], padding="VALID", strides=[1, 1, 2, 2], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0
/device:GPU:0"](mrcnn_mask_deconv/conv2d_transpose-0-VecPermuteNHWCToNCHW-LayoutOptimizer, mrcnn_mask_deconv/kernel/read, training/SGD/gradients/mrcnn_mask_deconv/conv2d_transpose_grad/Conv2DBackpropFilter-2-TransposeNHWCToNCHW-LayoutOptimizer)]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

         [[Node: training/SGD/gradients/mrcnn_class_conv1/convolution_grad/Conv2DBackpropInput-0-0-TransposeNCHWToNHWC-LayoutOptimizer/_2991 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replic
a:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_6826_...tOptimizer", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

I think that just means, that my memory has no space left right?
I am kind of new to this neuronal networking things and i didnt really find a tutorial or an explanation how to use the Amazon P2 things, thats why i thought it was only for linux(Found one link that says so), maybe you could help me out that would be really nice...

Edit: I found a tutorial how to setup a p2 instance, my question is now, how do i get all of my code onto the virtual machine so i can start it from there?

**Comment by marcfielding1** [comment]
So when you've created a p2 instance you need to download the SSH keys as prompted, the AWS docs cover that. 

Quick overview.
Launch your instance.
SSH into it.
Install/Enable SFTP.
Exit back to your machine
zip up your code.
Use SFTP to push it up to your p2 instance.
Go back to your P2 instance, extract and train!

**Comment by minalmeg** [comment]
https://github.com/matterport/Mask_RCNN/wiki here's a way to optimize your GPU. It's important to understand the tradeoff with accuracy before optimizing.

**Comment by burhr2** [comment]
Hi! You can also try this fresh implementation. kindly see this issue https://github.com/matterport/Mask_RCNN/issues/2312 there is a pointer to the same implementation that supports TensorFlow 2=> (with GPU).  If this solves your problem kindly close the issue so as others can navigate to other issues easier
